google-cloud-storage/src/test/java/com/google/cloud/storage/it/StorageNativeCanary.java

```diff
@@ -0,0 +1,182 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.it;
+
+import static com.google.cloud.storage.TestUtils.assertAll;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.ReadChannel;
+import com.google.cloud.WriteChannel;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.DataGenerator;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.Storage.BlobListOption;
+import com.google.cloud.storage.Storage.BlobSourceOption;
+import com.google.cloud.storage.Storage.BlobWriteOption;
+import com.google.cloud.storage.StorageOptions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteStreams;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
+import java.util.List;
+import java.util.UUID;
+import org.junit.Test;
+
+// Intentionally avoid StorageITRunner here. It touches lots of code at a semi-static level making
+// native-test have a hard time.
+public final class StorageNativeCanary {
+
+  private static final int _512KiB = 512 * 1024;
+  private static final int _256KiB = 256 * 1024;
+  private static final byte[] bytes = DataGenerator.base64Characters().genBytes(_512KiB);
+
+  @Test
+  public void canary_happyPath_http() throws Exception {
+    canary_happyPath(StorageOptions.http().build().getService());
+  }
+
+  @Test
+  public void canary_happyPath_grpc() throws Exception {
+    canary_happyPath(StorageOptions.grpc().build().getService());
+  }
+
+  /**
+   * When testing on Native Image, we're primarily wanting to verify the primary code paths are
+   * properly detected by the native image compiler.
+   *
+   * <p>For Storage, we have a few "primary code paths" we want to ensure are validated:
+   *
+   * <ul>
+   *   <li>Can a (Unary) Request Succeed?
+   *   <li>Can a (ServerStream) Object Read Request Succeed?
+   *   <li>Can a (ClientStream) Object Write Request Succeed?
+   *   <li>Can a (Page over Unary) Paginated Request Succeed?
+   * </ul>
+   *
+   * To validate this, our happy path test is as follows:
+   *
+   * <ul>
+   *   <li>Create a temporary bucket (Unary)
+   *   <li>Insert two (2) objects (Unary, ServerStream)
+   *   <li>List all objects, using a pageSize of 1 (Page over Unary)
+   *   <li>Read all bytes of each object (ServerStream)
+   *   <li>Delete each object (Unary)
+   *   <li>Delete temporary bucket (Unary)
+   * </ul>
+   */
+  private static void canary_happyPath(Storage storage) throws Exception {
```

Comment:
How about we rename this to something like `assertBehaviorOfPrimaryStorageActions`? Since the details of what we are testing are abstracted over by the helper method, the name can help us determine that.